### PR TITLE
[wip] Entrofy package

### PR DESCRIPTION
**Pull request by bmcfee**
For the 2016-04-11 sprint

To-dos:

- [ ]  Brian Data validation / diagnostics utility
- [x]  Brian tests for ObjectMapper
- [x]  Brian read-the-docs
- [x] Brian RNG
- [x] ~~Brian add `equals` option to ObjectMapper so users can supply their own equivalence calculator~~

---
This change is [Reviewable](https://reviewable.io/reviews/dhuppenkothen/entrofy/23)

### CODE REVIEW

**Comment by bmcfee** [comment]
To-dos:

- [ ] (Brian) Data validation / diagnostics utility
- [ ] (Brian) tests for ObjectMapper
- [ ] (Brian) ???

File: entrofy/mappers.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+        prefix: string, optional
+            A prefix string to go in front of the `column_names`
+
+        """
+
+        self.n_out = n_out
+        self.prefix = prefix
+
+        minval = np.nanmin(np.array(df))
```
Comment (bmcfee) [inline entrofy/mappers.py]: I think we need to specify that Mappers operate on series, not entire dataframes.

**Comment by bmcfee** [comment]
Upon further thought generalizing the equals comparator in `ObjectMapper` will be incompatible with #29 , since it would require serialization of lambda functions.  I think we should punt on it, and force users to pre-map their data if they have a particular comparison in mind.

**Comment by bmcfee** [comment]
I think validation / diagnostics can be punted out to [pandas-profiling](https://github.com/JosPolfliet/pandas-profiling)

**Comment by bmcfee** [comment]
Is this PR ready to merge?  I know we haven't finished the thing in any reasonable sense, but it seems like there's sufficient groundwork done already that we can merge and then start creating branches for individual features.

**Comment by dhuppenkothen** [comment]
Agreed. We should merge #30 first, though. :)

**Comment by bmcfee** [comment]
Right!  Is that one done/ready for review?

**Comment by dhuppenkothen** [comment]
Yes, I think that's ready for review.

**Comment by dhuppenkothen** [comment]
Merging this! Woo!

**Pull request merged**